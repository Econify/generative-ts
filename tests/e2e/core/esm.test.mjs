import * as esm from "../../../packages/core/dist/index.mjs";

describe('@generative-ts/core esm bundle', () => {
  test('api snapshot', async () => {
    expect(esm).toMatchSnapshot();
  });

  test('sanity check', async () => {
    const model = esm.createLmStudioModelProvider({
      modelId: "TheBloke/Mistral-7B-Instruct-v0.2-GGUF",
    });
    
    const response = await model.sendRequest({
      prompt: "Say hi:",
      max_tokens: 50,
      temperature: 0
    });

    expect(response.choices[0].message).toBeDefined();
  });
});
