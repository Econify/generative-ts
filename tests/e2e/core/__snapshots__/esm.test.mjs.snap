// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@generative-ts/core esm bundle api snapshot 1`] = `
{
  "Ai21Jurassic2Api": {
    "requestTemplate": h {
      "source": [Function],
    },
    "responseGuard": [Function],
  },
  "Ai21Jurassic2Template": h {
    "source": [Function],
  },
  "AmazonTitanTextApi": {
    "requestTemplate": h {
      "source": [Function],
    },
    "responseGuard": [Function],
  },
  "AmazonTitanTextTemplate": h {
    "source": [Function],
  },
  "BaseModelProvider": [Function],
  "CohereChatApi": {
    "name": "COHERE-CHAT",
    "requestTemplate": h {
      "source": [Function],
    },
    "responseGuard": [Function],
  },
  "CohereChatTemplate": h {
    "source": [Function],
  },
  "CohereGenerateApi": {
    "name": "COHERE-GENERATE",
    "requestTemplate": h {
      "source": [Function],
    },
    "responseGuard": [Function],
  },
  "CohereGenerateTemplate": h {
    "source": [Function],
  },
  "FnTemplate": [Function],
  "GoogleGeminiApi": {
    "requestTemplate": h {
      "source": [Function],
    },
    "responseGuard": [Function],
  },
  "GoogleGeminiTemplate": h {
    "source": [Function],
  },
  "HfConversationalTaskApi": {
    "requestTemplate": h {
      "source": [Function],
    },
    "responseGuard": [Function],
  },
  "HfConversationalTaskTemplate": h {
    "source": [Function],
  },
  "HfTextGenerationTaskApi": {
    "requestTemplate": h {
      "source": [Function],
    },
    "responseGuard": [Function],
  },
  "HfTextGenerationTaskTemplate": h {
    "source": [Function],
  },
  "HttpModelProvider": [Function],
  "Llama2ChatApi": {
    "requestTemplate": h {
      "source": [Function],
    },
    "responseGuard": [Function],
  },
  "Llama2ChatTemplate": h {
    "source": [Function],
  },
  "Llama3ChatApi": {
    "requestTemplate": h {
      "source": [Function],
    },
    "responseGuard": [Function],
  },
  "Llama3ChatTemplate": h {
    "source": [Function],
  },
  "MistralAiApi": {
    "requestTemplate": h {
      "source": [Function],
    },
    "responseGuard": [Function],
  },
  "MistralAiTemplate": h {
    "source": [Function],
  },
  "MistralBedrockApi": {
    "requestTemplate": h {
      "source": [Function],
    },
    "responseGuard": [Function],
  },
  "MistralBedrockTemplate": h {
    "source": [Function],
  },
  "OpenAiChatApi": {
    "requestTemplate": h {
      "source": [Function],
    },
    "responseGuard": [Function],
  },
  "OpenAiChatTemplate": h {
    "source": [Function],
  },
  "createAwsBedrockModelProvider": [Function],
  "createCohereModelProvider": [Function],
  "createGroqModelProvider": [Function],
  "createHuggingfaceInferenceModelProvider": [Function],
  "createLmStudioModelProvider": [Function],
  "createMistralModelProvider": [Function],
  "createOpenAiChatModelProvider": [Function],
}
`;
